custom_components/wyzeapi/cover.py

```diff
@@ -0,0 +1,188 @@
+        return True
+    
+    @property
+    def should_poll(self) -> bool:
+        return False
+    
+    @token_exception_handler
+    async def async_open_cover(self, **kwargs):
+        """Open the cover."""
+        try:
+            await self._camera_service.garage_door_open(self._camera)
+        except (AccessTokenError, ParameterError, UnknownApiError) as err:
+            raise HomeAssistantError(f"Wyze returned an error: {err.args}") from err
+        except Exception as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._camera.garage = True
+            self._out_of_sync_count = 0
+            self.async_write_ha_state()
+    
+    @token_exception_handler
+    async def async_close_cover(self, **kwargs):
+        """Close the cover."""
+        try:
+            await self._camera_service.garage_door_close(self._camera)
+        except (AccessTokenError, ParameterError, UnknownApiError) as err:
+            raise HomeAssistantError(f"Wyze returned an error: {err.args}") from err
+        except Exception as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._camera.garage = False
+            self._out_of_sync_count = 0
+            self.async_write_ha_state()
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        return self._camera.garage == False
+        
+    @property
+    def available(self):
+        """Return the connection status of this cover."""
+        return self._camera.available
+    
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE
+                
+    @property
+    def unique_id(self):
+        return f"{self._camera.mac}_GarageDoor"
+    @property
+    def name(self):
+        return "Garage Door"
+    
+    @token_exception_handler
+    async def async_update(self):
+        """Update the garage door status."""
+        camera = await self._camera_service.update(self._camera)
+
+        if camera.garage == self._camera.garage or MAX_OUT_OF_SYNC_COUNT <= self._out_of_sync_count:
+            self._out_of_sync_count = 0
+            self._camera = camera
+        else:
+            self._out_of_sync_count += 1
+
+    @callback
+    def async_update_callback(self, camera: Camera):
+        """Update the switch's state."""
+        self._camera = camera
+        async_dispatcher_send(
+            self.hass,
+            f"{COVER_UPDATED}-{self._camera.mac}",
+            garage,
+        )
+        self.async_schedule_update_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to update events."""
+        self._camera.callback_function = self.async_update_callback
+        self._camera_service.register_updater(self._camera, 10)
+        await self._camera_service.start_update_manager()
+        return await super().async_added_to_hass()
```

Comment:
```suggestion
    async def async_added_to_hass(self) -> None:
        self.async_on_remove(
            async_dispatcher_connect(
                self.hass,
                f"{CAMERA_UPDATED}-{self._device.mac}",
                self.handle_camera_update,
            )
        )
```

Reply:
So you'll want to test and make sure this works, but basically what's happening here is we don't need to add another updater to the update manager over in the other lib, because the camera is already being updated periodically. Instead we just listen for the camera to be updated, and then update this entity with that info. 

Take a look at how the siren or floodlights are set up.

Because of that I don't think we need the out of sync counter either

Reply:
I've just pushed this up. Will test with my HA shortly.

Reply:
aaand it works!